# Anova
Three archers – Pat, Jack, and Alex are participating in an archery contest. They are shooting at targets with 10 evenly spaced concentric rings. The rings have score values from 1 through 10 assigned to them, with 10 being the highest. Each participant shoots 6 arrow, scoring the following points: Pat – 5, 4, 4, 3, 9, 4. Jack – 4, 8, 7, 5, 1, 5. Alex – 9, 9, 8, 10, 4, 10 Based on the above results we would like to know who the best archer is. In other words our null hypothesis is that the means of all populations are equal.

In [1]:
import pandas as pd
import numpy as np
from scipy import stats


In [2]:
data = np.rec.array([('Pat',5),('Pat',4),('Pat',4),('Pat',3),('Pat',9),('Pat',4),('Jack',4),('Jack',8),('Jack',7),('Jack',5),
                    ('Jack',1),('Jack',5),('Alex',9),('Alex',9),('Alex',8),('Alex',10),('Alex',4),('Alex',10)], dtype = [('Archer','|U5'),('Score','<i8')])

In [3]:
f,p = stats.f_oneway(data[data['Archer'] == 'Pat'].Score,
                     data[data['Archer'] == 'Jack'].Score,
                     data[data['Archer'] == 'Alex'].Score)

In [4]:
print('One way ANOVA')
print(" ")
print('F value:',f)
print('P value:',p,'\n')

One way ANOVA
 
F value: 4.4882729211087415
P value: 0.029667763955533364 



In [5]:
from statsmodels.stats.multicomp import pairwise_tukeyhsd
from statsmodels.stats.multicomp import MultiComparison

In [6]:
mc = MultiComparison(data['Score'], data['Archer'])
result = mc.tukeyhsd()
 
print(result)
print(mc.groupsunique)

Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj   lower  upper  reject
---------------------------------------------------
  Alex   Jack  -3.3333 0.0568 -6.7554 0.0887  False
  Alex    Pat     -3.5 0.0447  -6.922 -0.078   True
  Jack    Pat  -0.1667    0.9 -3.5887 3.2554  False
---------------------------------------------------
['Alex' 'Jack' 'Pat']


The test also shows the difference between the group means(the meandiff column)

μ(jack) - μ(Alex) = -3.333 μ(pat) - μ(ALex) = -3.5

This leads to the conclusion that Alex is the best Archer in the gruop